codec-http2/src/main/java/io/netty/handler/codec/http2/CleartextHttp2ServerUpgradeHandler.java

```diff
@@ -89,8 +89,11 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) t
                 // following network traffic
                 ctx.pipeline()
                    .remove(httpServerCodec)
-                   .remove(httpServerUpgradeHandler)
-                   .replace(this, null, http2ServerHandler);
+                   .remove(httpServerUpgradeHandler);
+
+                ctx.pipeline().addAfter(ctx.name(), null, http2ServerHandler);
+                ctx.pipeline().remove(this);
+
                 ctx.fireUserEventTriggered(PriorKnowledgeUpgradeEvent.INSTANCE);
```

Comment:
In netty itself, no. But maybe the api user would like to know there is an upgrade event fired.